vsphere/panels.libsonnet

```diff
@@ -0,0 +1,1118 @@
+        + barGauge.queryOptions.withTargets([
+          t.clusterMemoryLimit,
+        ])
+        + barGauge.panelOptions.withDescription('The available memory capacity of the cluster.')
+        + barGauge.options.withOrientation('horizontal')
+        + barGauge.standardOptions.withUnit('decbytes')
+        + barGauge.standardOptions.color.withMode('continuous-YlRd'),
+
+      clusterMemoryUtilization:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Cluster memory utilization',
+          targets=[t.clusterMemoryUtilization],
+          description='The memory utilization percentage of the cluster.'
+        )
+        + g.panel.timeSeries.standardOptions.withUnit('percent')
+        + g.panel.timeSeries.options.legend.withDisplayMode('table'),
+
+      esxiHostsTable:
+        commonlib.panels.generic.table.base.new(
+          'ESXi hosts table',
+          targets=[
+            t.hostCPUUtilizationCluster + g.query.prometheus.withFormat('table')
+            + g.query.prometheus.withRange(true)
+            ,
+            t.hostMemoryUtilizationCluster + g.query.prometheus.withFormat('table')
+            + g.query.prometheus.withRange(true)
+            ,
+            t.networkThroughputRateCluster + g.query.prometheus.withFormat('table')
+            + g.query.prometheus.withRange(true)
+            ,
+            t.hostCPUUsageCluster + g.query.prometheus.withFormat('table')
+            + g.query.prometheus.withRange(true)
+            ,
+            t.hostMemoryUsageCluster + g.query.prometheus.withFormat('table')
+            + g.query.prometheus.withRange(true),
+          ]
+          ,
+          description='A table displaying information about the ESXi hosts in the vCenter environment.'
+        )
+        + table.standardOptions.withOverridesMixin([
+          fieldOverride.byName.new('CPU utilization')
+          + fieldOverride.byName.withProperty('custom.displayMode', 'gradient-gauge')
+          + fieldOverride.byName.withProperty('custom.align', 'left')
+          + fieldOverride.byName.withProperty('custom.minWidth', 250)
+          + fieldOverride.byName.withPropertiesFromOptions(
+            table.standardOptions.withUnit('percent')
+            + table.standardOptions.color.withMode('continuous-BlPu')
+            + table.standardOptions.withMin(0)
+            + table.standardOptions.withMax(100)
+            + table.standardOptions.withDecimals(1)
+          ),
+        ])
+        + table.standardOptions.withOverridesMixin([
+          fieldOverride.byName.new('Network throughput')
+          + fieldOverride.byName.withProperty('custom.align', 'left')
+          + fieldOverride.byName.withPropertiesFromOptions(
+            table.standardOptions.withUnit('KiBs')
+          ),
+        ])
+        + table.standardOptions.withOverridesMixin([
+          fieldOverride.byName.new('CPU usage')
+          + fieldOverride.byName.withProperty('custom.align', 'left')
```

Comment:
This header is still right justified, unless the cloud dashboard isn't updated.

Reply:
The cloud dashboard wasn't fully updated with this change. It should be good.